libp2p/RLPxHandshake.cpp

```diff
@@ -112,10 +112,12 @@ void RLPXHandshake::writeAckEIP8()
     m_ackCipher.insert(m_ackCipher.begin(), prefix.begin(), prefix.end());
     
     auto self(shared_from_this());
-    ba::async_write(m_socket->ref(), ba::buffer(m_ackCipher), [this, self](boost::system::error_code ec, std::size_t)
-    {
-        transition(ec);
-    });
+    ba::async_write(m_socket->ref(), ba::buffer(m_ackCipher),
+        [this, self](boost::system::error_code ec, std::size_t) {
+            if (ec)
```

Comment:
It looks like we can put `if (ec) m_failureReason = TcpError;` inside `transition(ec)` and avoid some repetition.

Reply:
Good point 😄